libs/s25main/figures/nofMiner.cpp

```diff
@@ -71,20 +72,100 @@ helpers::OptionalEnum<GoodType> nofMiner::ProduceWare()
 
 bool nofMiner::AreWaresAvailable() const
 {
+    // FindPointWithResource triggeres outofresource message
+    if(GetAddonSetting() == 3)
+        return true;
+
     return nofWorkman::AreWaresAvailable() && FindPointWithResource(GetRequiredResType()).isValid();
 }
 
-bool nofMiner::StartWorking()
+unsigned int nofMiner::GetAddonSetting() const 
 {
-    MapPoint resPt = FindPointWithResource(GetRequiredResType());
-    if(!resPt.isValid())
-        return false;
     const GlobalGameSettings& settings = world->GetGGS();
-    bool inexhaustibleRes = settings.isEnabled(AddonId::INEXHAUSTIBLE_MINES)
-                            || (workplace->GetBuildingType() == BuildingType::GraniteMine
-                                && settings.isEnabled(AddonId::INEXHAUSTIBLE_GRANITEMINES));
-    if(!inexhaustibleRes)
-        world->ReduceResource(resPt);
+
+    switch(workplace->GetBuildingType())
+    {
+        case BuildingType::GoldMine: return settings.getSelection(AddonId::MINES_GOLD);
+        case BuildingType::IronMine: return settings.getSelection(AddonId::MINES_IRON);
+        case BuildingType::CoalMine: return settings.getSelection(AddonId::MINES_COAL);
+        case BuildingType::GraniteMine: return settings.getSelection(AddonId::MINES_GRANITE);
+        default: return 0;
+    }
+}
+
+bool nofMiner::StartWorking()
+{
+    workplace->is_emptyCycle = false;
```

Comment:
Why is this a property of the workplace? Shouldn't this be part of the miner? You can use this in `ProduceWare` to return `boost::none` aka "nothing"